x-pack/plugins/spaces/server/lib/get_spaces_usage_collector.ts

```diff
@@ -12,29 +15,78 @@ import { KIBANA_SPACES_STATS_TYPE } from '../../common/constants';
  * @param callCluster
  * @param server
  * @param {boolean} spacesAvailable
- * @param withinDayRange
- * @return {ReportingUsageStats}
+ * @return {UsageStats}
  */
-async function getSpacesUsage(callCluster: any, server: any, spacesAvailable: boolean) {
+async function getSpacesUsage(callCluster: any, server: Server, spacesAvailable: boolean) {
   if (!spacesAvailable) {
-    return {};
+    return {} as UsageStats;
   }
 
-  const { getSavedObjectsRepository } = server.savedObjects;
+  const index = server.config().get('kibana.index');
+
+  const knownFeatureIds = server.plugins.xpack_main.getFeatures().map(feature => feature.id);
+
+  const resp = await callCluster('search', {
+    index,
+    body: {
+      track_total_hits: true,
+      query: {
+        term: {
+          type: {
+            value: 'space',
+          },
+        },
+      },
+      aggs: {
+        disabledFeatures: {
+          terms: {
+            field: 'space.disabledFeatures',
+            size: knownFeatureIds.length,
```

Comment:
Spaces with unknown disabled features have the ability to "push out" known disabled features with the current bucket size. For example, if there are three known features: `feature1`, `feature2`, and `feature3`, but there is a space with `unknownFeature1` disabled, then it's possible for the result to look something like:

```
{
 aggregations: {
    disabledFeatures: {
       buckets: [{
          key: 'feature1',
          doc_count: 1
       }, {
          key: 'feature2',
          doc_count: 3
       }, {
          key: 'unknownFeature1',
          doc_count: 1
       }]
    }
 }
}
```

There might be a space with `feature3` disabled, but we don't have results because of the terms size specified here.  We can mitigate this by increasing the size (maybe `knownFeatureIds.length * 2`?), but I don't know of a way to completely guard against this. Thoughts?

Reply:
Hmm, what if we were to only report on the disabled features which are also known features?

Or, perhaps we could use https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-cardinality-aggregation.html to determine the count first?

Reply:
> Hmm, what if we were to only report on the disabled features which are also known features?

We don't report on the unknown features - they end up getting filtered out [here](https://github.com/elastic/kibana/pull/35348/files#diff-622ce95aab51e3bf5aadfd587cdc14feR59).

> Or, perhaps we could use elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-cardinality-aggregation.html to determine the count first?

That's not a bad idea. I also just came across [Composite Aggregations](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-composite-aggregation.html), which seems to fit our use case:
```
If you want to retrieve all terms or all combinations of terms in a nested
terms aggregation you should use the Composite aggregation which allows to
paginate over all possible terms rather than setting a size greater than the
cardinality of the field in the terms aggregation. The terms aggregation is
meant to return the top terms and does not allow pagination.
```

Reply:
I do wonder if the added complexity of either approach is worth the outcome though. The odds of this happening seem pretty slim to me, unless an API user is dumping garbage into the `spaces.disabledFeatures` property.

Reply:
Even better: [Terms Aggregation supports an `include` parameter](https://www.elastic.co/guide/en/elasticsearch/reference/6.7/search-aggregations-bucket-terms-aggregation.html#_filtering_values_3), which accepts an array of values you want to bucket on. This will prevent the multiple round-trips and added complexity of our earlier ideas.

Reply:
See https://github.com/elastic/kibana/pull/35348/commits/f65d942a86f18213a256e126d183f93cc6186a50 for implementation using `include`